### DISCUSSION TITLE
Get the complete discographies from all artists in a playlist

### DISCUSSION THREAD

**Discussion by ffmusic** [Chimildic/goofy#147] [Помощь с алгоритмом] [answered]
Hi everyone
I've been trying to do this without success so far.
Let's suppose I have a small playlist with 5 songs, each one from a different artist.
Goofy would grab a list of all the artists from that playlist (in this case 5) and export their complete discographies to another playlist.
Can someone help?
Thanks

**Comment by Chimildic** [comment, marked as answer]
Just insert playlist id and run

```js
function createDiscographyByArtists() {
  let tracks = Source.getPlaylistTracks('', 'id');
  let discographies = Source.getArtistsTracks({
    artist: {
      followed_include: false,
      include: tracks.map(t => t.artists[0]),
    },
    album: {
      groups: 'album,single',
    },
  });

  Filter.dedupTracks(discographies);
  Playlist.saveWithReplace({
    name: 'Discographies',
    tracks: discographies,
    randomCover: 'update',
  });
}
```

**Reply by ffmusic** [reply]
Thanks a lot, Chimildic, it works flawlessly!
Helpful as always.
